processor/integrations/integrations.go

```diff
@@ -38,111 +54,29 @@ const (
-type PostParameterT struct {
-	URL           string
-	Type          int
-	UserID        string
-	Payload       interface{}
-	Header        interface{}
-	RequestConfig interface{}
-}
-
-// PostParameterNewT emulates parameters needed tp make a request
-type PostParameterNewT struct {
-	Type          string
-	URL           string
-	RequestMethod string
-	UserID        string
-	Headers       interface{}
-	QueryParams   interface{}
-	Body          interface{}
-	Files         interface{}
-}
-
-// GetResponseVersion Get version of the transformer response
-func GetResponseVersion(response json.RawMessage) string {
-	parsedResponse := gjson.ParseBytes(response)
-	if parsedResponse.Get("output").Exists() {
-		return "-1"
-	}
-	if !parsedResponse.Get("version").Exists() {
-		return "0"
-	}
-	version, ok := parsedResponse.Get("version").Value().(string)
-	if !ok {
-		panic(fmt.Errorf(""))
-	}
-	return version
-}
-
-// GetPostInfoNew parses the transformer response
-func GetPostInfoNew(transformRaw json.RawMessage) PostParameterNewT {
-	var postInfo PostParameterNewT
-	var ok bool
-	parsedJSON := gjson.ParseBytes(transformRaw)
-	if parsedJSON.Get("output").Exists() {
-		parsedJSON = parsedJSON.Get("output")
-	}
-	postInfo.Type, ok = parsedJSON.Get("type").Value().(string)
-	if !ok {
-		panic(fmt.Errorf("typecast of parsedJSON.Get(\"type\") to string failed"))
-	}
-	postInfo.URL, ok = parsedJSON.Get("endpoint").Value().(string)
-	if !ok {
-		panic(fmt.Errorf("typecast of parsedJSON.Get(\"endpoint\") to string failed"))
-	}
-	postInfo.RequestMethod, ok = parsedJSON.Get("method").Value().(string)
-	if !ok {
-		panic(fmt.Errorf("typecast of parsedJSON.Get(\"method\") to string failed"))
-	}
-	postInfo.UserID, ok = parsedJSON.Get("userId").Value().(string)
-	if !ok {
-		postInfo.UserID = fmt.Sprintf("%v", parsedJSON.Get("userId").Value())
-	}
-	postInfo.Body, ok = parsedJSON.Get("body").Value().(interface{})
-	if !ok {
-		panic(fmt.Errorf("typecast of parsedJSON.Get(\"body\") to interface{} failed"))
-	}
-	postInfo.Headers, ok = parsedJSON.Get("headers").Value().(interface{})
-	if !ok {
-		panic(fmt.Errorf("typecast of parsedJSON.Get(\"headers\") to interface{} failed"))
-	}
-	postInfo.QueryParams, ok = parsedJSON.Get("params").Value().(interface{})
-	if !ok {
-		panic(fmt.Errorf("typecast of parsedJSON.Get(\"params\") to interface{} failed"))
-	}
-	postInfo.Files, ok = parsedJSON.Get("files").Value().(interface{})
-	if !ok {
-		panic(fmt.Errorf("typecast of parsedJSON.Get(\"files\") to interface{} failed"))
-	}
-	return postInfo
+// PostParametersT is a struct for holding all the values from transformerResponse and use them to publish an event to a destination 
+type PostParametersT struct {
+	Type          string      `json:"type"`
+	URL           string      `json:"endpoint"`
+	RequestMethod string      `json:"method"`
+	UserID        string      `json:"userId"`
+	Headers       interface{} `json:"headers"`
```

Comment:
Make all these `map[string]interface{}` except files?

Reply:
I wasn't entirely sure if we can do it because this typecast is happening in it's caller and that too only conditionally.

Reply:
We can do this. It makes things more readable. Please change it.